### PR TITLE
Reorganize Data Depot controllers to components

### PR DESCRIPTION
**Pull request by jarosenb**
Restructure controller/template pairs in the data depot into components.

This is just code reorganization without any fancy ES6 class stuff. The only real change is that some state from the projects controllers has been moved to the projects service, because scope can't be shared between components like we were doing between controllers.

### CODE REVIEW

File: designsafe/static/scripts/data-depot/components/community/community.component.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import _ from 'underscore';
+import CommunityTemplate from './community.component.html';
+
+export function CommunityDataCtrl($scope, $state, Django, DataBrowserService) {
```
Comment (xirdneh) [inline designsafe/static/scripts/data-depot/components/community/community.component.js]: Add
```
'ngInject';
```

File: designsafe/static/scripts/data-depot/components/my-data/my-data.component.js [hunk 2]
```diff
@@ -1,20 +1,8 @@
 import _ from 'underscore';
+import MyDataTemplate from './my-data.component.html';
+
+export function MyDataCtrl($scope, $state, Django, DataBrowserService) {
```
Comment (xirdneh) [inline designsafe/static/scripts/data-depot/components/my-data/my-data.component.js]: Add `'ngInject';`

File: designsafe/static/scripts/data-depot/components/projects/project-root/project-root.component.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import _ from 'underscore';
+import projectRootTemplate from './project-root.component.html';
+
+export function ProjectRootCtrl($scope, $state, DataBrowserService, ProjectService) {
+    $scope.browser = DataBrowserService.state();
```
Comment (xirdneh) [inline designsafe/static/scripts/data-depot/components/projects/project-root/project-root.component.js]: Add `'ngInject';`

File: designsafe/static/scripts/data-depot/components/projects/project-search/project-search.component.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import ProjectSearchTemplate from './project-search.component.html';
+import _ from 'underscore';
+
+function ProjectSearchCtrl($scope, $state, Django, DataBrowserService, ProjectService) {
+
```
Comment (xirdneh) [inline designsafe/static/scripts/data-depot/components/projects/project-search/project-search.component.js]: Add `'ngInject';`

File: designsafe/static/scripts/data-depot/components/community/community.component.js (hunk 1 above)
Comment (jarosenb) [reply designsafe/static/scripts/data-depot/components/community/community.component.js]: I've added all the `ngInject` annotations and made sure it works with `npm run build`. 

Can you clarify what you mean by "add comments on the controllers to deprecate the use of $scope" ? Those controllers still use $scope (that will hopefully change soon though), I only replaced $scope variables with service variables when state was being shared between components.

Comment (xirdneh) [reply designsafe/static/scripts/data-depot/components/community/community.component.js]: I mean to add an explicit comment that we need to do that. But i think by now we all know we have to do it. So, it's ok :)

**Pull request merged**